app/lib/system_name.rb

```diff
@@ -26,37 +28,37 @@ def has_system_name(opts = {})
 
   protected
 
-    def system_human_name
-      send(self._system_human_name)
-    end
+  def system_human_name
+    send(_system_human_name)
+  end
 
-    # Generates system_name from 'system_human_name'
-    def generate_system_name
-      generate_system_name! if self.system_name.blank?
-    end
+  # Generates system_name from 'system_human_name'
+  def generate_system_name
+    generate_system_name! if system_name.blank?
+  end
 
-    def generate_system_name!
-      if self.system_human_name.present?
-        slug = self.system_human_name.parameterize.underscore
-        self.system_name = slug.blank? ? internal_system_name : slug
-      else
-        # TODO: remove this branch when Feature#name validation is added
-        # and this ticket resolved https://github.com/3scale/system/issues/908
-        if self.is_a?(Feature)
-          msg = "Deprecated API with missing feature name probably called."
-          Rails.logger.warn msg
-        end
-
-        self.system_name = internal_system_name
+  def generate_system_name!
+    if system_human_name.present?
+      slug = system_human_name.parameterize.underscore
+      self.system_name = slug.presence || internal_system_name
+    else
+      # TODO: remove this branch when Feature#name validation is added
+      # and this ticket resolved https://github.com/3scale/system/issues/908
```

Comment:
This probably can already be cleaned but I won't blame u if u decide that it is out of scope :smile: But then I would appreciate creating a Jira issue about this and writing the link here in a comment :stuck_out_tongue:

Reply:
Added as comment: https://issues.redhat.com/browse/THREESCALE-4179